AsyncDisplayKit/ASControlNode.mm

```diff
@@ -424,5 +436,80 @@ - (void)cancelTrackingWithEvent:(UIEvent *)touchEvent
 - (void)endTrackingWithTouch:(UITouch *)touch withEvent:(UIEvent *)touchEvent
 {
 }
+#if TARGET_OS_TV
+#pragma mark - tvOS
+- (void)pressDown
+{
+  [UIView animateWithDuration:0.1 delay:0 options:UIViewAnimationCurveLinear animations:^{
+    [self setPressedState];
+  } completion:^(BOOL finished) {
+    if (finished) {
+      [UIView animateWithDuration:0.1 delay:0 options:UIViewAnimationCurveLinear animations:^{
+        [self setFocusedState];
+      } completion:nil];
+    }
+  }];
+}
 
+- (BOOL)canBecomeFocused
+{
+  return YES;
+}
+
+- (BOOL)shouldUpdateFocusInContext:(nonnull UIFocusUpdateContext *)context
+{
+  return YES;
+}
+
+- (void)didUpdateFocusInContext:(UIFocusUpdateContext *)context withAnimationCoordinator:(UIFocusAnimationCoordinator *)coordinator
+{
+  //FIXME: This is never valid inside an ASCellNode
+  if (context.nextFocusedView && context.nextFocusedView == self.view) {
+    //Focused
+    [coordinator addCoordinatedAnimations:^{
+      [self setFocusedState];
+    } completion:nil];
+  } else{
+    //Not focused
+    [coordinator addCoordinatedAnimations:^{
+      [self setDefaultState];
+    } completion:nil];
+  }
+}
+
+- (void)setFocusedState
+{
+  CALayer *layer = self.layer;
+  layer.shadowOffset = CGSizeMake(2, 10);
+  [self applyDefaultShadowProperties: layer];
+  self.view.transform = CGAffineTransformScale(CGAffineTransformIdentity, 1.1, 1.1);
+}
+
+- (void)setPressedState
+{
+  CALayer *layer = self.layer;
+  layer.shadowOffset = CGSizeMake(2, 2);
+  [self applyDefaultShadowProperties: layer];
+  self.view.transform = CGAffineTransformScale(CGAffineTransformIdentity, 1, 1);
+}
+
+- (void)applyDefaultShadowProperties:(CALayer *)layer
+{
+  layer.shadowColor = [UIColor blackColor].CGColor;
+  layer.shadowRadius = 12.0;
+  layer.shadowOpacity = 0.45;
+  layer.shadowPath = [UIBezierPath bezierPathWithRect:self.layer.bounds].CGPath;
+}
+
+- (void)setDefaultState
+{
+  CALayer *layer = self.layer;
+  layer.shadowOffset = CGSizeZero;
+  layer.shadowColor = [UIColor blackColor].CGColor;
+  layer.shadowRadius = 0;
+  layer.shadowOpacity = 0;
+  layer.shadowPath = [UIBezierPath bezierPathWithRect:self.layer.bounds].CGPath;
+  self.view.transform = CGAffineTransformScale(CGAffineTransformIdentity, 1, 1);
+}
+#endif
```

Comment:
For longer blocks, put // TARGET_OS_TV at the #endif to help clarify what it is ending